cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java

```diff
@@ -323,6 +324,49 @@ private static void writeGenerators(Network network, Map<Terminal, String> mapTe
         }
     }
 
+    private static void writeBatteries(Network network, String cimNamespace, boolean writeInitialP,
+                                        XMLStreamWriter writer, CgmesExportContext context) throws XMLStreamException {
+        // Multiple synchronous machines may be grouped in the same generating unit
+        // We have to write each generating unit only once
+        Set<String> generatingUnitsWritten = new HashSet<>();
+        for (Battery battery : network.getBatteries()) {
+            String generatingUnit = context.getNamingStrategy().getCgmesIdFromProperty(battery, Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + "GeneratingUnit");
+            String reactiveLimitsId = null;
+            double minQ = 0.0;
+            double maxQ = 0.0;
+            switch (battery.getReactiveLimits().getKind()) {
+                case CURVE:
+                    reactiveLimitsId = CgmesExportUtil.getUniqueId();
+                    ReactiveCapabilityCurve curve = battery.getReactiveLimits(ReactiveCapabilityCurve.class);
+                    for (ReactiveCapabilityCurve.Point point : curve.getPoints()) {
+                        CurveDataEq.write(CgmesExportUtil.getUniqueId(), point.getP(), point.getMinQ(), point.getMaxQ(), reactiveLimitsId, cimNamespace, writer, context);
+                    }
+                    String reactiveCapabilityCurveName = "RCC_" + battery.getNameOrId();
+                    ReactiveCapabilityCurveEq.write(reactiveLimitsId, reactiveCapabilityCurveName, battery, cimNamespace, writer, context);
+                    break;
+
+                case MIN_MAX:
+                    minQ = battery.getReactiveLimits(MinMaxReactiveLimits.class).getMinQ();
+                    maxQ = battery.getReactiveLimits(MinMaxReactiveLimits.class).getMaxQ();
+                    break;
+
+                default:
+                    throw new PowsyblException("Unexpected type of ReactiveLimits on the generator " + battery.getNameOrId());
+            }
+            SynchronousMachineEq.write(context.getNamingStrategy().getCgmesId(battery), battery.getNameOrId(),
+                    context.getNamingStrategy().getCgmesId(battery.getTerminal().getVoltageLevel()),
+                    generatingUnit, null, reactiveLimitsId, minQ, maxQ, Double.NaN, cimNamespace, writer, context);
+            if (!generatingUnitsWritten.contains(generatingUnit)) {
```

Comment:
the block that writes the generating unit is also common to generators and batteries; move it to a separate function to avoid duplications

Reply:
Refactoring done